lib/src/utils/rest_args_with_separator.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:args/args.dart';
+
+/// Returns the "rest" args from [argResults], but with the arg separator "--"
+/// restored to its original position if it was included.
+///
+/// This is necessary because [ArgResults.rest] will _not_ include the separator
+/// unless it stopped parsing before it reached the separator.
+///
+/// The use case for this is a [CompoundTool] that uses the [takeAllArgs] arg
+/// mapper, because the goal there is to forward on the original args minus the
+/// consumed options and flags. If the separator has also been removed, you may
+/// hit an error when trying to parse those args.
+///
+///     var parser = ArgParser()..addFlag('verbose', abbr: 'v');
+///     var results = parser.parse(['a', '-v', 'b', '--', '--unknown', 'c']);
+///     print(results.rest);
+///     // ['a', 'b', '--unknown', 'c']
+///     print(restArgsWithSeparator(results));
+///     // ['a', 'b', '--', '--unknown', '-c']
+List<String> restArgsWithSeparator(ArgResults argResults) {
+  // If no separator was used, return the rest args as is.
+  if (!argResults.arguments.contains('--')) {
+    return argResults.rest;
+  }
+
+  final args = argResults.arguments;
+  final rest = argResults.rest;
+  var rCursor = 0;
```

Comment:
Maybe a clearer name? e.g. argSeparatorPosition

Reply:
I'm a little torn on this. I originally had something like what you suggested, but then usages like this feel a bit more confusing:

```
if (args[aCursor] == rest[separatorArgPosition]) {
```

Reply:
Yeah, I see that. What about not abbreviating. argsCursor and restCursor. I see aCursor at a glance and read it as "an instance of Cursor". Or argsIndex and restIndex?